### ISSUE TITLE
android.database.CursorWindowAllocationException  in  LiveChatUtil.java line 444 

### ISSUE THREAD

**Issue by alokvnair** [zoho/salesiq-mobilisten-android-sample#25]
Getting below error in Zoho Mobilisten SDK. SDK version used in 2.3.1. 

```
Fatal Exception: android.database.CursorWindowAllocationException: Could not allocate CursorWindow '/data/user/0/com.noonEdu.k12App/databases/siq_mobilisten.db' of size 4194304 due to error -12.
       at android.database.CursorWindow.nativeCreate(CursorWindow.java)
       at android.database.CursorWindow.<init>(CursorWindow.java:145)
       at android.database.sqlite.SQLiteCursor.clearOrCreateWindow(SQLiteCursor.java:319)
       at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:159)
       at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:152)
       at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:232)
       at android.database.AbstractCursor.moveToFirst(AbstractCursor.java:271)
       at com.zoho.livechat.android.utils.LiveChatUtil.getLastMessage(LiveChatUtil.java:444)
       at com.zoho.livechat.android.ui.fragments.ChatFragment.canEnableSendButton(ChatFragment.java:3470)
       at com.zoho.livechat.android.ui.fragments.ChatFragment.setSendInputButtonState(ChatFragment.java:3410)
       at com.zoho.livechat.android.ui.fragments.ChatFragment.afterTextChanged(ChatFragment.java:3368)
       at android.widget.TextView.sendAfterTextChanged(TextView.java:11664)
       at android.widget.TextView$ChangeWatcher.afterTextChanged(TextView.java:14910)
       at android.text.SpannableStringBuilder.sendAfterTextChanged(SpannableStringBuilder.java:1277)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:577)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:507)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:37)
       at android.view.inputmethod.BaseInputConnection.replaceText(BaseInputConnection.java:849)
       at android.view.inputmethod.BaseInputConnection.commitText(BaseInputConnection.java:199)
       at com.android.internal.widget.EditableInputConnection.commitText(EditableInputConnection.java:177)
       at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:339)
       at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:89)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:7948)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1075)Fatal Exception: android.database.CursorWindowAllocationException: Could not allocate CursorWindow '/data/user/0/com.noonEdu.k12App/databases/siq_mobilisten.db' of size 4194304 due to error -12.
       at android.database.CursorWindow.nativeCreate(CursorWindow.java)
       at android.database.CursorWindow.<init>(CursorWindow.java:145)
       at android.database.sqlite.SQLiteCursor.clearOrCreateWindow(SQLiteCursor.java:319)
       at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:159)
       at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:152)
       at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:232)
       at android.database.AbstractCursor.moveToFirst(AbstractCursor.java:271)
       at com.zoho.livechat.android.utils.LiveChatUtil.getLastMessage(LiveChatUtil.java:444)
       at com.zoho.livechat.android.ui.fragments.ChatFragment.canEnableSendButton(ChatFragment.java:3470)
       at com.zoho.livechat.android.ui.fragments.ChatFragment.setSendInputButtonState(ChatFragment.java:3410)
       at com.zoho.livechat.android.ui.fragments.ChatFragment.afterTextChanged(ChatFragment.java:3368)
       at android.widget.TextView.sendAfterTextChanged(TextView.java:11664)
       at android.widget.TextView$ChangeWatcher.afterTextChanged(TextView.java:14910)
       at android.text.SpannableStringBuilder.sendAfterTextChanged(SpannableStringBuilder.java:1277)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:577)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:507)
       at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:37)
       at android.view.inputmethod.BaseInputConnection.replaceText(BaseInputConnection.java:849)
       at android.view.inputmethod.BaseInputConnection.commitText(BaseInputConnection.java:199)
       at com.android.internal.widget.EditableInputConnection.commitText(EditableInputConnection.java:177)
       at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:339)
       at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:89)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:7948)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1075)
```

<img width="1150" alt="Screenshot 2021-02-09 at 4 00 57 PM" src="https://user-images.githubusercontent.com/8415897/107351046-275c7c80-6af0-11eb-850a-6249efe17cfd.png">

**Comment by Angu-G** [comment]
@alokvnair, Thank you for your patience. We have found the root cause for this issue and the similar one #24. Fix will be rolled out in our next immediate update.

**Comment by alokvnair** [comment]
Thanks @Angu-G for the update. Please do update here once the release is made. Also any update on [https://github.com/zoho/salesiq-mobilisten-android-sample/issues/23](#23)

**Comment by Angu-G** [comment]
Yeah, we are working on that one. Will let you know by today.

**Comment by Angu-G** [comment]
Hai @alokvnair, Thank you for your patience. We have a fix for this issue in our latest version [v3.0](https://github.com/zoho/salesiq-mobilisten-android-sample/releases/tag/3.0). Please try it out and let us know. Closing the issue.
